packages/playwright-core/src/server/frames.ts

```diff
@@ -268,7 +268,7 @@ export class FrameManager {
   }
 
   requestStarted(request: network.Request, route?: network.RouteDelegate) {
-    const frame = request.frame();
+    const frame = request.frame()!;
```

Comment:
I'd make this method call page.requestStarted that calls context.requestStarted so that we get a nice chain of methods that always bubble up.

Reply:
I think this might still be relevant.

Reply:
Plan of action:
- inline `page._requestStarted` into `frameManager._onRequestStarted`;
- remove `browserContext._requestStarted` and inline it into `CRServiceWorker._requestStarted`.
We end up with two separate implementations.

Reply:
This should now be fixed.